File: software/examples/delphi/ExampleSimple.pas
program ExampleSimple;

{$ifdef MSWINDOWS}{$apptype CONSOLE}{$endif}
{$ifdef FPC}{$mode OBJFPC}{$H+}{$endif}

uses
  SysUtils, IPConnection, BrickletTilt;

type
  TExample = class
  private
    ipcon: TIPConnection;
    t: TBrickletTilt;
  public
    procedure Execute;
  end;

const
  HOST = 'localhost';
  PORT = 4223;
  UID = 'XYZ'; { Change XYZ to the UID of your Tilt Bricklet }

var
  e: TExample;

procedure TExample.Execute;
var state: byte;
begin
  { Create IP connection }
  ipcon := TIPConnection.Create;

  { Create device object }
  t := TBrickletTilt.Create(UID, ipcon);

  { Connect to brickd }
  ipcon.Connect(HOST, PORT);
  { Don't use device before ipcon is connected }

  { Get current tilt state }
  state := t.GetTiltState;

  case state of
    BRICKLET_TILT_TILT_STATE_CLOSED:
    begin
      WriteLn('Tilt State: Closed');
    end;
    BRICKLET_TILT_TILT_STATE_OPEN:
    begin
      WriteLn('Tilt State: Open');
    end;
    BRICKLET_TILT_TILT_STATE_CLOSED_VIBRATING:
    begin
      WriteLn('Tilt State: Closed Vibrating');
    end;
  end;

  WriteLn('Press key to exit');
  ReadLn;
  ipcon.Destroy; { Calls ipcon.Disconnect internally }
end;

begin
  e := TExample.Create;
  e.Execute;
  e.Destroy;
end.

File: software/examples/vbnet/ExampleSimple.vb
Imports System
Imports Tinkerforge

Module ExampleSimple
    Const HOST As String = "localhost"
    Const PORT As Integer = 4223
    Const UID As String = "XYZ" ' Change XYZ to the UID of your Tilt Bricklet

    Sub Main()
        Dim ipcon As New IPConnection() ' Create IP connection
        Dim t As New BrickletTilt(UID, ipcon) ' Create device object

        ipcon.Connect(HOST, PORT) ' Connect to brickd
        ' Don't use device before ipcon is connected

        ' Get current tilt state
        Dim state As Byte = t.GetTiltState()

        If state = BrickletTilt.TILT_STATE_CLOSED Then
            Console.WriteLine("Tilt State: Closed")
        Else If state = BrickletTilt.TILT_STATE_OPEN Then
            Console.WriteLine("Tilt State: Open")
        Else If state = BrickletTilt.TILT_STATE_CLOSED_VIBRATING Then
            Console.WriteLine("Tilt State: Closed Vibrating")
        End If

        Console.WriteLine("Press key to exit")
        Console.ReadLine()
        ipcon.Disconnect()
    End Sub
End Module
